strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java

```diff
@@ -103,21 +112,27 @@ public boolean isMavenMetadata(RepositoryPath path)
             switch (attributeType)
             {
                 case ARTIFACT:
-                    value = (Boolean) value && !isMavenMetadata(repositoryPath) && !isIndex(repositoryPath);
+                    value = BooleanUtils.isTrue((Boolean) value) && !isMavenMetadata(repositoryPath) &&
+                            !isIndex(repositoryPath);
 
-                    if (value != null)
-                    {
-                        result.put(attributeType, value);
-                    }
+                    result.put(attributeType, value);
 
                     break;
                 case METADATA:
-                    value = (Boolean) value || isMavenMetadata(repositoryPath);
+                    value = BooleanUtils.isTrue((Boolean) value) || isMavenMetadata(repositoryPath);
 
-                    if (value != null)
-                    {
-                        result.put(attributeType, value);
-                    }
+                    result.put(attributeType, value);
+
+                    break;
+                case EXPIRED:
+                    value = BooleanUtils.isTrue((Boolean) value) || (isMavenMetadata(repositoryPath)
+                                                                     &&
+                                                                     !RepositoryFiles.wasModifiedAfter(repositoryPath,
+                                                                                                       Instant.now()
```

Comment:
this probably better to be extracted into variable

Reply:
Yeah, thanks